src/listener.py

```diff
@@ -0,0 +1,190 @@
+			# The destination is ME!
+			# if ip_layer.fields["protocol"].get()
+
+		# If the packet need to be forward
+		elif destination_ip in ROUTING_TABLE["net1"].keys() or destination_ip in ROUTING_TABLE["net2"].keys():
+
+
+			network = get_network_by_ip(IPAddress.ip2str(destination_ip))
+
+			ethernet_layer.fields["src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+			ethernet_layer.fields["dst"].set(MacAddress.str2mac(ROUTING_TABLE[network][destination_ip]))
+
+			buffer = buffer[14:]
+
+			buffer = ethernet_layer.build() + buffer
+
+			PACKETS_TO_SEND.append((find_network_place(network) , buffer))
+
+			print("Packet forward")
+		else:
+			return
+
+
+
+def parse_ARP(buffer, ethernet_layer):
+	#ARP part
+	arp_layer = ArpLayer()
+	arp_layer.deserialize(buffer[:28])
+
+	destination_ip = IPAddress.ip2str(arp_layer.fields["ip_dst"].get())
+
+	if destination_ip == MY_ADDRESSES["net1"][1] or destination_ip == MY_ADDRESSES["net2"][1]:
+
+		src_ip = IPAddress.ip2str(arp_layer.fields["ip_src"].get())
+		
+		if not verify_addresses(src_ip, MacAddress.mac2str(arp_layer.fields["mac_src"].get())):
+			return
+
+		if arp_layer.fields["opcode"].get() == ArpLayer.OP_WHO_HAS:
+			# ethernet_layer.connect_layer(arp_layer)
+			build_ARP_response(ethernet_layer, arp_layer)
+
+
+def build_ARP_response(ethernet_layer, arp_layer):
+	global PACKETS_TO_SEND
+
+	network = get_network_by_ip(IPAddress.ip2str(arp_layer.fields["ip_src"].get()))
+
+	# Switch the source and destination MAC addresses.
+	ethernet_layer.fields["dst"].set(ethernet_layer.fields["src"].get())
+
+	ethernet_layer.fields["src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+	# Change the addresses in the ARP Layer and the opcode.
+	arp_layer.fields["opcode"].set(ArpLayer.OP_IS_AT)
+
+	arp_layer.fields["ip_dst"].set(arp_layer.fields["ip_src"].get())
+
+	arp_layer.fields["mac_dst"].set(arp_layer.fields["mac_src"].get())
+
+	arp_layer.fields["mac_src"].set(MacAddress.str2mac(MY_ADDRESSES[network][0]))
+
+	arp_layer.fields["ip_src"].set(IPAddress.str2ip(MY_ADDRESSES[network][1]))
+
+	ethernet_layer.connect_layer(arp_layer)
+
+	PACKETS_TO_SEND.append((find_network_place(network) , ethernet_layer.build()))
+
+	print("ARP response sent")
+
+
+
+def create_sockets():
+
+	read_sockets = []
+	read_sockets.append(socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.htons(socket.PACKET_OTHERHOST)))
+	read_sockets[0].bind(('net1', 0))
+	read_sockets.append(socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.htons(socket.PACKET_OTHERHOST)))
+	read_sockets[1].bind(('net2', 0))
+	return read_sockets
+
+def find_network_place(network):
+	return list(ROUTING_TABLE.keys()).index(network)
+
+def main():
+	global PACKETS_TO_SEND
+	read_sockets = create_sockets()
+
+	while True:
+		#rlist - clients we want and can to read data from
+		#wlist - clients we want and can to send data to them
+		#xlist - clients that have a error
+		rlist, wlist, xlist = select.select(read_sockets, read_sockets, [])
```

Comment:
Do you do anything with `wlist`? If not, why pass something to the 2nd argument of `select`?